app/code/Magento/Eav/Model/Mview/BatchIterator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Eav\Mview;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Sql\Expression;
+use Magento\Framework\Mview\View\ChangeLogBatchIteratorInterface;
+use Magento\Framework\Mview\View\ChangelogInterface;
+use Magento\Framework\Mview\View\ChangelogTableNotExistsException;
+use Magento\Framework\Phrase;
+
+/**
+ * Class BatchIterator
+ */
+class BatchIterator implements ChangeLogBatchIteratorInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection
+    ) {
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function walk(ChangelogInterface $changelog, int $fromVersionId, int $toVersion, int $batchSize)
+    {
+        $connection = $this->resourceConnection->getConnection();
+        if (!$connection->isTableExists($changelog->getName())) {
```

Comment:
Why we need this check. Query will fail in any case

Reply:
It was copied fromm the old code. I don`t why